marketplace_standard_app_api/main.py

```diff
@@ -0,0 +1,245 @@
+from typing import Any, Callable, Dict, Optional
+
+import requests
+from fastapi import Depends, FastAPI, HTTPException, Request
+from fastapi.responses import HTMLResponse, JSONResponse, Response
+
+from .models import (
+    DatasetCreateResponse,
+    DatasetId,
+    DatasetModel,
+    NewTransformationModel,
+    TransformationCreateResponse,
+    TransformationId,
+    TransformationListResponse,
+    TransformationModel,
+    TransformationStatusResponse,
+    TransformationUpdateModel,
+    TransformationUpdateResponse,
+)
+from .security import AuthTokenBearer
+from .version import __version__
+
+
+async def catch_authentication_request_errors_middleware(
+    request: Request, call_next: Callable
+):
+    "Catch authentication requests errors to the semantic service and respond with 401."
+    try:
+        return await call_next(request)
+    except requests.exceptions.HTTPError as error:
+        if error.response.status_code == 401:
+            return Response("Not authenticated.", status_code=401)
+        raise
+
+
+class MarketPlaceAPI(FastAPI):
+    def openapi(self) -> Dict[str, Any]:
+        openapi_schema = super().openapi()
+        openapi_schema["info"]["x-application-name"] = "MarketPlace Template App"
+        return openapi_schema
+
+
+api = MarketPlaceAPI(
+    title="MarketPlace Standard App API",
+    description="Standard app API for the MarketPlace applications.",
+    version=__version__,
+    contact={
+        "name": "The Materials MarketPlace Consortium",
+        "url": "https://www.materials-marketplace.eu/",
+        "email": "dirk.helm@iwm.fraunhofer.de",
+    },
+    license_info={"name": "MIT", "url": "https://opensource.org/licenses/MIT"},
+    dependencies=[Depends(AuthTokenBearer())],
+)
+api.middleware("http")(catch_authentication_request_errors_middleware)
+
+
+@api.get(
+    "/",
+    operation_id="frontend",
+    tags=["FrontPage"],
+    responses={
+        404: {"description": "Not found."},
+        401: {"description": "Not authenticated."},
+        500: {"description": "Internal server error."},
+        503: {"description": "Service unavailable."},
+    },
+    response_class=HTMLResponse,
+)
+async def frontpage() -> HTMLResponse:
+    """Open the frontpage of the app."""
+    raise HTTPException(status_code=404, detail="Not found.")
```

Comment:
Indeed, we could use 501 as well here.

Reply:
Thinking about it again, I think that using 501 instead of 404 does not make much sense here. This particular operation must return an HTML page. There are only two possible scenarios in my view, either that page exists or it does not. In case it exists it is returned and we use code 200. If it does not exists we can decide to send 404 or 501.

With 404 we indicate that the page does not exist and we cannot return it. With 501 we indicate that we never intended to return one. However, how does that differ from a user point of view. What additional meaning is conveyed in this way? I feel like 404 is the appropriate response when requesting an HTML page that does not exist.

@pablo-de-andres @yoavnash What do you think?

Reply:
After offline discussion, we agree to use 501.